### PR TITLE
DOC: Docstring examples for cogent3.app.sample.trim_stop_codons

### LINKED ISSUE #1754: Add code example snippet to cogent3.app.sample.trim_stop_codons
The code should be executable by doctesting. See `omit_degenerates()`, `omit_gap_pos()`, `concat()` docstrings for examples. The purpose of these examples is a user gets help on an app and the help displays working code. In other words, the sample code you write should work in an interactive interpreter once the preceding `>>> ` have been removed. It should also not load any data from file, but create sequence collections (in this case) from dictionaries with small sequences.

This app removes trailing stop codons from sequences in a sequence collection or alignment. The output will vary based on whether the input is a sequence collection or an alignment.

For examples, see the `test_trim_stop_codons()` method in `tests/test_app/test_sample.py`. You should only use a minimal data set to demonstrate how it works, not the entire data set used for testing.

### PR DESCRIPTION
**Pull request by fredjaya**
Edits to @Lizhaozhe123 contributions on PR https://github.com/cogent3/cogent3/pull/1767.

Fixes #1754.

### CODE REVIEW

File: src/cogent3/app/sample.py [hunk 1]
```diff
@@ -820,18 +820,46 @@ def main(self, seqs: SeqsCollectionType) -> T:
...
+        collection or alignment.
+
+        Create a sample unaligned sequence data and an app to remove trailing
+        stop codons.
+
+        >>> from cogent3 import make_unaligned_seqs, make_aligned_seqs, get_app
+        >>> ualn = make_unaligned_seqs(data={"s1": "AAATTTCCC", "s2": "AAATTTTAA"}, moltype="dna")
+        >>> app = get_app("trim_stop_codons")
+        >>> result = app(ualn)
+        >>> print(result.to_dict())
```
Comment (GavinHuttley) [inline src/cogent3/app/sample.py:849]: change to pretty_print()

File: src/cogent3/app/sample.py [hunk 2]
```diff
@@ -820,18 +820,46 @@ def main(self, seqs: SeqsCollectionType) -> T:
...
+        Remove trailing stop codons from aligned sequences, using the genetic
+        code for "Vertebrate Mitochondrial".
+
+        For a list of all genetic codes, see
+        https://cogent3.org/doc/cookbook/what_codes.html.
+
+        >>> aln = make_aligned_seqs(data={"s1": "AAATTTCCC", "s2": "AAATTTTAA"}, moltype="dna")
+        >>> app = get_app("trim_stop_codons", gc=2)
+        >>> result = app(aln)
+        >>> print(result.to_dict())
```
Comment (GavinHuttley) [inline src/cogent3/app/sample.py:861]: as above

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
small tweaks

File: src/cogent3/app/sample.py (hunk 1 above)
Comment (fredjaya) [reply src/cogent3/app/sample.py:849]: `AttributeError: 'SequenceCollection' object has no attribute 'to_pretty'`. Works for the next example where it's aligned. Suggestions?

Comment (fredjaya) [reply src/cogent3/app/sample.py:849]: `AttributeError: 'SequenceCollection' object has no attribute 'to_pretty'`. Works for the next example that returns an `Aligned` instance. Suggestions?

Comment (GavinHuttley) [reply src/cogent3/app/sample.py:849]: Nah, you're right. `to_dict()` is fine.

**Pull request merged**